less/_navbar.less

```diff
@@ -188,13 +188,47 @@
   .generic-variations(~".navbar", ~"", @brand-primary, {
     background-color: @variation-color;
     color: @variation-color-text;
-    // deeply defined to override welljumbo class without !impotant need
-    .navbar-form .form-group input.form-control,
-    .navbar-form input.form-control {
-      .material-placeholder({
-        color: @variation-color-text;
-      });
+
+    .navbar-form {
+      input.form-control,
+      .form-group input.form-control {
+        .material-placeholder({
+          color: @variation-color-text;
+        });
+      }
+
+      .form-group {
+        &.label-floating,
+        &.label-placeholder {
+          label.control-label {
+            color: @variation-color-text;
+            text-overflow: ellipsis;
+            overflow: hidden;
+            white-space: nowrap;
+            opacity: 0.87;
```

Comment:
The majority of this change should not be inside a .generic-variations call.  This creates 4 or so color variations.  If it isn't overriding the color, it should not be in this file.